File: KAKAO_BLIND_RECRUITMENT/2020/문자열_압축/solve.cpp
#include <algorithm>
#include <iostream>
#include <vector>
#include <queue>

using namespace std;

int LENGTH = 0;
int MIN_LENGTH;

int solution(string S){
    ios_base :: sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    LENGTH = S.length();
    MIN_LENGTH = LENGTH;

    for(int i=1; i<=LENGTH; i++){
        string result = "";
        queue<string> que;
        
        string cur = "";
        for(int j=0; j<LENGTH; j++){
            if(cur.length() == i){
                que.push(cur);
                cur = "";
            }

            cur += S[j];
        }
        
        if(cur.size() >= 1){
            que.push(cur);
        }
        
        string prev = que.front();
        que.pop();
        int cnt = 1;
        
        while(!que.empty()){
            string now = que.front();
            que.pop();
            
            if(prev == now){
                cnt++;
            } else {
                if(cnt == 1) result += prev;
                else result += to_string(cnt) + prev;
                prev = now;
                cnt = 1;
            }
        }
        
        if(cnt == 1) result += prev;
        else result += to_string(cnt) + prev;
        
        MIN_LENGTH = min(MIN_LENGTH, (int)result.length());
    }
    
    return MIN_LENGTH;
    
}

File: 탐욕법/단속카메라/solve.py
def solution(routes):
    routes.sort()
    
    end = routes[0][1]
    
    cnt = 1
    for i in range(1, len(routes)):
        # 시작 지점이 end지점보다 작은 경우
        if routes[i][0] <= end:
            end = min(end, routes[i][1])
        # 그 외는 도착 지점을 end로 최신화 
        else:
            cnt += 1
            end = routes[i][1]
            
    return cnt
